src/btree/bt_handle.c

```diff
@@ -995,18 +995,16 @@ __btree_page_sizes(WT_SESSION_IMPL *session)
 bool
 __wt_btree_immediately_durable(WT_SESSION_IMPL *session)
 {
-    WT_BTREE *btree;
-
-    btree = S2BT(session);
-
     /*
-     * This is used to determine whether timestamp updates should be rolled back for this btree.
-     * With in-memory, the logging setting on tables is still important and when enabled they should
-     * be considered "durable".
+     * If timestamp updates should be rolled back for this btree in RTS as well as configuring some
+     * checkpoint and open behaviors.
+     *
+     * Historically, this function checked for in-memory configurations as well, assert that's not
+     * the case for now.
```

Comment:
Fixed in 474a05e5c. This isn't a temporary change, and I did it because it didn't make sense to ask if an in-memory configuration is "immediately durable", there is no durability for in-memory configs.